### ISSUE TITLE
Remove raw HTML string interpolation

### ISSUE THREAD

**Issue by saschanaz** [w3c/respec#2551]
One consensus in #2548 is that we should remove raw string interpolations. Now we can easily remove them since the PR found all the raw things 🎉

* *Where are those interpolations?:* Find-on-page by `raw(` and all of them are raw string interpolations. There are 21 matches as of now.
* *How to remove them?*: Convert them to inline `Node` or `Element` elements, e.g. `` html`<var>${str}</var>` ``

**Comment by saschanaz** [comment]
@marcoscaceres Candidate for good first issue? 👀

**Comment by marcoscaceres** [comment]
I think so! We just need to make them super easy to find, and really clear how to fix them.

**Comment by saschanaz** [comment]
Do we want to keep `conf.*HTML` e.g. `conf.wgHTML` or not?

**Comment by marcoscaceres** [comment]
I don't think so... those are all part of the "override" tag'ed tasks.

**Comment by saschanaz** [comment]
>I don't think so... those are all part of the "override" tag'ed tasks.

You mean you want to remove the overriding things? GitHub search shows only [one use of `wgHTML`](https://github.com/OpenMobileAlliance/Tools-Old/blob/031d1d4cb2a49bd201c9a1f82bd8a50592d332cb/oma-profile/oma/headers.js) which is marked as old 👀

**Comment by marcoscaceres** [comment]
Maybe we can just entirely remove that. 

What I mean is, for a lot of things that insert html from `config`, authors should be actually writing those things directly in the document. There are whole bunch of these:
https://github.com/w3c/respec/issues?q=is%3Aissue+is%3Aopen+label%3AOverride

**Comment by saschanaz** [comment]
Right, just like how additional SOTD contents are processed.

**Comment by deepesh-ludhyani** [comment]
I would be glad to try to solve this issue.

**Comment by saschanaz** [comment]
Hello @deepesh-ludhyani, I'm assigning you for this issue then! Feel free to ask questions whenever needed. 👍

**Comment by SuyashSalampuria** [comment]
@saschanaz  if no one is working on this, shall I try to solve this issue

**Comment by saschanaz** [comment]
@deepesh-ludhyani Are you working on this?

**Comment by SuyashSalampuria** [comment]
@saschanaz  I am starting to work on this issue.
Please tell if I have understood the issue correctly
Is the requirement to convert
<li>
      ${testFileName}
      <a href="${href}">
    </a> ${emojiList}
        ${testFileName}
      </a> ${emojiList}
    </li>

to 
<li>${testFileName}<a href="${href}"></a> ${emojiList}${testFileName}</a> ${emojiList}</li>

**Comment by SuyashSalampuria** [comment]
@saschanaz I am starting to work on this issue.
Please tell if I have understood the issue correctly
Is the requirement to convert
<li>
      <a href="${href}">
        ${testFileName}
      </a> ${emojiList}
    </li>
to
<li><a href="${href}">${testFileName}</a> ${emojiList}</li>

**Comment by SuyashSalampuria** [comment]
@saschanaz  I am starting to work on this issue
Can you confirm if I have understood the issue correctly
Please confirm if have to convert
![Screenshot from 2019-11-26 23-40-25](https://user-images.githubusercontent.com/44131389/69660474-6ac2fa80-10a6-11ea-89f2-7ea590a4a69c.png)
to 
```<li>  <a href="${href}">  ${testFileName} </a> ${emojiList} </li>```

**Comment by saschanaz** [comment]
@SuyashSalampuria No, that one is fine. What do we want is that we replace normal JavaScript strings into hyperHTML templated string so that we get all of its feature including escaping.

**Comment by marcoscaceres** [comment]
@saschanaz, as this is security critical, I'd be more comfortable if you were working on it.

**Comment by SuyashSalampuria** [comment]
I have tried my best to understand the problem and solve it. 
Although,  this was my first contribution to open source. Please tell me if I have understood the problem correctly and solved it as required

**Comment by ridhishjain** [comment]
@saschanaz, can I take this issue?

**Comment by saschanaz** [comment]
Several are already removed by others, not sure whether there is a remaining. If you can find one, please do.

**Comment by ridhishjain** [comment]
I did not find any in src/w3c,  src/ui and, src/geonovum
any other directories to look for?
otherwise, the issue is resolved I guess

**Comment by marcoscaceres** [comment]
This is done. All raw HTML string interpolations have been removed from the codebase (zero instances of `raw(` in `src/`). As @ridhishjain-zepto confirmed in their last comment, the cleanup was completed. Thanks to everyone who contributed.
